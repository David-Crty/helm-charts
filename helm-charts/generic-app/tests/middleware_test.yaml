# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: test cronjob
release:
  name: test
chart:
  version: 0.1.0
templates:
  - middleware.yaml
  - ingress.yaml
tests:
  - it: should have a redirect middleware
    set:
      ingress:
        enabled: true
        hosts:
          - crty.dev
          - www.crty.dev
          - infra.crty.dev
        redirections:
          - from:
              - www.crty.dev
              - infra.crty.dev
            to: crty.dev
    asserts:
      - matchSnapshot: {}