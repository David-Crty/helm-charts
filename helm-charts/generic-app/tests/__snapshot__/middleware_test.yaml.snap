should have a redirect middleware:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: letsencrypt-prod
        traefik.ingress.kubernetes.io/router.middlewares: NAMESPACE-test-generic-app-redirect-to-crty-dev@kubernetescrd,traefik-redirect-to-https@kubernetescrd
      name: test-generic-app
    spec:
      ingressClassName: traefik
      rules:
        - host: crty.dev
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
        - host: www.crty.dev
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
        - host: infra.crty.dev
          http:
            paths:
              - backend:
                  service:
                    name: test-generic-app
                    port:
                      number: 80
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - crty.dev
            - www.crty.dev
            - infra.crty.dev
          secretName: test-generic-app-cert
  2: |
    apiVersion: traefik.io/v1alpha1
    kind: Middleware
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app-redirect-to-crty-dev
    spec:
      redirectRegex:
        permanent: true
        regex: ^https?://(www\.crty\.dev|infra\.crty\.dev)(/.*)
        replacement: https://crty.dev$2
